actors/init/src/lib.rs

```diff
@@ -62,19 +64,33 @@ impl Actor {
 
         log::trace!("called exec; params.code_cid: {:?}", &params.code_cid);
 
-        let caller_code = rt.get_actor_code_cid(&rt.message().caller()).ok_or_else(|| {
-            actor_error!(illegal_state, "no code for caller as {}", rt.message().caller())
+        // TODO figure out ACLs -- allow exec anything for now
+        // let caller_code = rt.get_actor_code_cid(&rt.message().caller()).ok_or_else(|| {
+        //     actor_error!(illegal_state, "no code for caller as {}", rt.message().caller())
+        // })?;
+
+        // log::trace!("caller code CID: {:?}", &caller_code);
+
+        // if !can_exec(rt, &caller_code, &params.code_cid) {
+        //     return Err(actor_error!(forbidden;
+        //             "called type {} cannot exec actor type {}",
+        //             &caller_code, &params.code_cid
+        //     ));
+        // }
+
+        // Ensure the code is installed/loaded
+        rt.transaction(|st: &mut State, rt| {
+            if st.is_installed_actor(rt.store(), &params.code_cid).map_err(|e| {
+                e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to check state")
+            })? {
+                rt.install_actor(&params.code_cid).map_err(|e| {
+                    e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to check state")
+                })
```

Comment:
I don't understand this bit? Is this re-installing the actor if it's already installed?

Reply:
yeah, we need to somehow ensure it is loaded -- we might have restarted since we first installed it.